convert2rhel/unit_tests/cert_test.py

```diff
@@ -16,77 +16,109 @@
-                utils.DATA_DIR = os.path.join(self.base_data_dir, rhel_version, arch)
-                system_cert = cert.SystemCert()
-                cert_path = system_cert._source_cert_path
-                self.assertEqual(cert_path, "{0}/rhel-certs/{1}".format(utils.DATA_DIR, pem))
-
-    @unit_tests.mock(cert, "loggerinst", unit_tests.GetLoggerMocked())
-    @unit_tests.mock(utils, "DATA_DIR", unit_tests.TMP_DIR)
-    @unit_tests.mock(system_info, "arch", "arch")
-    def test_get_cert_path_missing_cert(self):
         # Create temporary directory that has no certificate
-        cert_dir = os.path.join(utils.DATA_DIR, "rhel-certs", system_info.arch)
+        cert_dir = os.path.join(rhel_cert_dir, request.param["arch"])
         utils.mkdir_p(cert_dir)
-        # Check response for the non-existing certificate in the temporary dir
-        self.assertRaises(SystemExit, cert.SystemCert._get_cert)
-        self.assertEqual(len(cert.loggerinst.critical_msgs), 1)
-        # Remove the temporary directory tree
+
+    monkeypatch.setattr(utils, "DATA_DIR", data_dir)
+    monkeypatch.setattr(system_info, "arch", request.param["arch"])
+
+    yield rhel_cert_dir
+
+    # Remove the temporary directory tree if we created it earlier
+    if request.param["data_dir"] is not None:
         shutil.rmtree(os.path.join(utils.DATA_DIR, "rhel-certs"))
 
-    @unit_tests.mock(cert, "loggerinst", unit_tests.GetLoggerMocked())
-    @unit_tests.mock(utils, "DATA_DIR", unit_tests.NONEXISTING_DIR)
-    @unit_tests.mock(system_info, "arch", "nonexisting_arch")
-    def test_get_cert_path_nonexisting_dir(self):
-        self.assertRaises(SystemExit, cert.SystemCert._get_cert)
-        self.assertEqual(len(cert.loggerinst.critical_msgs), 1)
-
-    @unit_tests.mock(tool_opts, "arch", "x86_64")
-    @unit_tests.mock(utils, "DATA_DIR", os.path.join(base_data_dir, "6", "x86_64"))
-    def test_install_cert(self):
-        # By initializing the cert object we get a path to an existing
-        # certificate based on the mocked parameters above
-        system_cert = cert.SystemCert()
-        system_cert._target_cert_dir = unit_tests.TMP_DIR
-
-        system_cert.install()
-
-        installed_cert_path = os.path.join(system_cert._target_cert_dir, system_cert._cert_filename)
-        self.assertTrue(os.path.exists(installed_cert_path))
-        shutil.rmtree(unit_tests.TMP_DIR)
+
+@pytest.mark.parametrize(
+    (
+        "message",
+        "cert_dir",
+    ),
+    (
+        # Missing certificate
+        (
+            "Error: System certificate (.pem) not found in %(cert_dir)s.",
+            {
+                "data_dir": unit_tests.TMP_DIR,
+                "arch": "arch",
+            },
+        ),
+        # Missing cert dir
+        (
+            "Error: Could not access %(cert_dir)s.",
+            {
+                "data_dir": None,
+                "arch": "nonexisting_arch",
+            },
+        ),
```

Comment:
Have you considered giving an `id` to the test param? This way you don't need to rely on a comment in the code, but actually have it as a named id (I think it shows under the test run too) 

```suggestion
        pytest.param(
            "Error: System certificate (.pem) not found in %(cert_dir)s.",
            {
                "data_dir": unit_tests.TMP_DIR,
                "arch": "arch",
            },
            id="missing-certificate",
        ),
        pytest.param(
            "Error: Could not access %(cert_dir)s.",
            {
                "data_dir": None,
                "arch": "nonexisting_arch",
            },
            id="missing-cert-dir",
        ),
```

Reply:
Nice!  I didn't know about that feature.  I like it a lot!

Reply:
I remembered about it the other day, so I thought it might come in handy for those types of tests that we need to specify what is going on. Probably it would be a good idea to use the same logic in the `test_hide_utils` at some point to…